src/migration_runner.rs

```diff
@@ -188,6 +188,20 @@ impl MigrationRunner {
         Ok(())
     }
 
+    pub fn apply_schema_snapshot(&self, schema: &str) -> anyhow::Result<()> {
+        let mut tx = self.pool.start_transaction(self.tx_opts)?;
+        for command in schema.split(";\n") {
```

Comment:
This assumption gives me pause, but I can't come up with a case where semicolon-newline would appear and wouldn't delineate statements.